### PR TITLE
WT-4629 Adding test coverage for oldest reader, all committed and pin…

### PR DESCRIPTION
**Pull request by luke-pearson**
…ned timestaps

### CODE REVIEW

**Comment by luke-pearson** [comment]
restest this please

**Comment by luke-pearson** [comment]
retest this please

File: test/suite/test_timestamp14.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        # a greater timestamp already existing.
+        self.assertTimestampsEqual(
+            self.conn.query_timestamp('get=all_committed'), "3")
+
+        # Senario 3: Commit with an all committed of 5 and then begin a
+        # transaction intending to commit at 4, the all_committed timestamp
+        # should move back.
+        session1.begin_transaction()
+        cur1[1] = 3
+        session1.commit_transaction('commit_timestamp=5')
```
Comment (bvpvamsikrishna) [inline test/suite/test_timestamp14.py]: commit timestamp is inconsistent with the comment, code is inconsistent with scenario specified above.

File: test/suite/test_timestamp14.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        session2.begin_transaction()
+        cur2[2] = 2
+        session2.commit_transaction('commit_timestamp=6')
+
+        self.assertTimestampsEqual(
+            self.conn.query_timestamp('get=all_committed'), "6")
+        cur1[1] = 2
+        session1.commit_transaction()
+
+    def test_oldest_reader(self):
```
Comment (bvpvamsikrishna) [inline test/suite/test_timestamp14.py]: can add some scenario's to use set_timestamp API for oldest timestamp and check the behavior.

File: test/suite/test_timestamp14.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+import random
+from suite_subprocess import suite_subprocess
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+def timestamp_str(t):
+    return '%x' % t
+
+class test_timestamp14(wttest.WiredTigerTestCase, suite_subprocess):
```
Comment (bvpvamsikrishna) [inline test/suite/test_timestamp14.py]: I prefer to have some scenario's which cover all three timestamps.

**Comment by luke-pearson** [comment]
retest this please

**Comment by luke-pearson** [comment]
retest this please

**Pull request merged**